docs/nv2/README.md

```diff
@@ -0,0 +1,376 @@
+  file:hello-world_v1-manifest.json
+```
+
+The formatted x509, without the `x5c` chain signature: `hello-world.signature.config.json` is:
+
+
+```json
+{
+    "signed": {
+        "iat": 1595456231,
+        "manifests": [
+            {
+                "digest": "sha256:407a722870b09ef1c037b3bd9d1e6fa828a1c64964ba8c292a8ebe4dcf3bde56",
+                "size": 3056,
+                "references": [
+                    "acme-rockets.io/hello-world:v1"
+                ]
+            }
+        ]
+
+    },
+    "signatures": [
+        {
+            "typ": "x509",
+            "sig": "FyRRcAGi1qd0IHT8XoRh0vlGSkE4rjYpJzYEjodRQ2aUO0O/bBIzjV86UxqLnb1Y/GMU817YXeqHqDlySLWYoGKg+/aJGJDqbQpWIxUr6hhjGaxBYDZTt2ayzMAu5X/GNCx0vRLKl5dOOsgTO53QbuKEf4F3xxvQJv3rXHnObJbPSPCavxzs5TNRLepYEZzW1Mp5nkZT4l32/7QLnwwzTsJYGOMTmhGZ7O5LB/eeViKmwBJHXpNzd4rytFXccKlPuyUakSKgsPdTjEvY5UbFpH568wG21HXDQivz6qdST9eSVob2yUx7WV7z+2S2GfmiMZ30BMtKs4Jx1uPOY3Hk8g==",
+            "alg": "RS256",
+            "kid": "2MKO:CS4G:GP3F:HELH:TUI2:5YSX:NJNU:3O2N:LYM4:FBHC:T7NN:OM5A"
+
+        }
+    ]
+}
+```
+
+Within the signature, the claims  `alg`, `x5c`, `kid` are specified by [RFC 7515](https://tools.ietf.org/html/rfc7515)
+
+### Signing using GnuPG
+
+To sign the manifest `example.json` using the GnuPG key identified by the identity name `Demo User`, run
+
+``` shell
+nv2 sign -m gpg \
+  -i "Demo User" \
+  -r registry.acme-rockets.io/hello-world:v1 \
+  -e 8760h \
+  -o hello-world.signature.config.json \
+  file:hello-world_v1-manifest.json
+```
+
+- `-r` declares the original registry reference - which must match the key.
+- `-e` specifies the optional expiry time (`8760h = 365 days`)
+
+On successful signing, `nv2` prints out the `sha256` digest of the manifest, and writes the `nv2` signature JSON file `<digest>.nv2` to the working directory. If the file name is not desired, option `-o` can be specified for the alternative file name.
+
+The formatted signature file is:
+
+```json
+{
+  "signed": {
+    "exp": 1626792407,
+    "nbf": 1595256407,
+    "iat": 1595256407,
+    "manifests": [
+      {
+        "digest": "sha256:3351c53952446db17d21b86cfe5829ae70f823aff5d410fbf09dff820a39ab55",
+        "size": 528,
+        "references": [
+          "example.registry.io/example:latest"
+        ]
+      }
+    ]
+  },
+  "signatures": [
+    {
+      "typ": "gpg",
+      "sig": "wsDcBAABCAAQBQJfFa5XCRBGnsnNNqPeHgAAlM0MAHFXZxyCgsxiGVat8YCRIhR7IoQe2scswGyvGGinYBy88EpKGFEAGO+Kt1frTQNW9kLYWmTw4EFctgMw+XxeDD/CI2rsMSluRh9h2t8xBsO9Ux+7eJoxSEsfU8Jc/YZWpGs/kJOGQ3ERjvPt+SCG0Y8tuNtjnzpV4Gz+8fLSlNZ7b3f+rd7nvvJuB8iWr+yojsCeWh/VGuibyqAXPKVxSrKgkmziyYK3O/0D3KhgyR+CMtjTXL5hP314Gpc415YyN82LC3L44okimN/+X3avX0vQkthiyVw+R+Vgmpa1qk1P/ySrs81yQgFBPBC7+m4n54TqsW46X/UlkQdfP/x5Jg3jUURKgQb0wSLvzbr7Jk1RiThlwjcLhM0VgRIUwbqcqjg/5UNvMRehD44PxQXRz5feZjER2awMyKqRZnImpm8Ub+hAjhqtLGYT34oU2lwctoObV4f4BzffY9kQ0x37PQ3V8aj8k6YFQZbB4vLgwtZdA2c1froVHyuRBUwLzSBevg==",
+      "iss": "Demo User <demo@example.com>"
```

Comment:
Is this issuing UID used at all or is this for readability? Also, I don't think there's anything in the JWT RFC that talks about encoding rfc4880 signatures inside of RFC7519 payloads.

In fact, things are a little bit more confusing because RFC4880 uses [issuer for the keyid octets](https://tools.ietf.org/html/rfc4880#section-5.2.3.5). A more appropriate way to do this would be to have RFC4880 subpackets for [Signing User ID](https://tools.ietf.org/html/rfc4880#section-5.2.3.22) extracted from the signature payload and inserted here.

Reply:
It is the UID of the signer. The main purpose of the `iss` is for readability.